spring-webclient-mocking/src/test/java/com/baeldung/webclientmocking/service/EmployeeServiceTest.java

```diff
@@ -0,0 +1,120 @@
+
+        StepVerifier.create(employeeMono)
+                .expectNextMatches(employee -> employee.getRole().equals(Role.LEAD_ENGINEER))
+                .verifyComplete();
+    }
+
+    @Test
+    void givenEmployee_whenAddEmployee_thenAddNewEmployee(){
+
+        Employee newEmployee = new Employee(null, "Adam", "Sandler", 32, Role.LEAD_ENGINEER);
+        Employee webClientResponse = new Employee(100, "Adam", "Sandler", 32, Role.LEAD_ENGINEER);
+        when(webClient.post()).thenReturn(requestBodyUriSpec);
+        when(requestBodyUriSpec.uri(EmployeeService.ADD_EMPLOYEE)).thenReturn(requestBodySpec);
+        when(requestBodySpec.syncBody(newEmployee)).thenReturn(requestHeadersSpec);
+        when(requestHeadersSpec.retrieve()).thenReturn(responseSpec);
+        when(responseSpec.bodyToMono(Employee.class)).thenReturn(Mono.just(webClientResponse));
+
+        Mono<Employee> employeeMono = employeeService.addNewEmployee(newEmployee);
+
+        StepVerifier.create(employeeMono)
+                .expectNextMatches(employee -> employee.getEmployeeId().equals(100))
+                .verifyComplete();
+    }
+
+    @Test
+    void givenEmployee_whenupdateEmployee_thenUpdatedEmployee(){
+
+        Integer employeeId=100;
+        Integer newAge=33;
+        String newLastName="Sandler New";
+        Employee updateEmployee = new Employee(100, "Adam", newLastName, newAge, Role.LEAD_ENGINEER);
+        when(webClient.put()).thenReturn(requestBodyUriSpec);
+        when(requestBodyUriSpec.uri(EmployeeService.PATH_PARAM_BY_ID,employeeId)).thenReturn(requestBodySpec);
+        when(requestBodySpec.syncBody(updateEmployee)).thenReturn(requestHeadersSpec);
+        when(requestHeadersSpec.retrieve()).thenReturn(responseSpec);
+        when(responseSpec.bodyToMono(Employee.class)).thenReturn(Mono.just(updateEmployee));
+
+        Mono<Employee> updatedEmploye =employeeService.updateEmployee(employeeId, updateEmployee);
+
+        StepVerifier.create(updatedEmploye)
+                .expectNextMatches(employee -> employee.getLastName().equals(newLastName) && employee.getAge() == newAge)
+                .verifyComplete();
+
+    }
+
+    @Test
+    void givenEmployee_whenDeleteEmployeeById_thenDeleteSuccessful(){
+
+        String responseMessage = "Employee Deleted SuccessFully";
+        Integer employeeId=100;
+        when(webClient.delete()).thenReturn(requestHeadersUriSpec);
+        when(requestHeadersUriSpec.uri(EmployeeService.PATH_PARAM_BY_ID, employeeId)).thenReturn(requestHeadersSpec);
+        when(requestHeadersSpec.retrieve()).thenReturn(responseSpec);
+        when(responseSpec.bodyToMono(String.class)).thenReturn(Mono.just(responseMessage));
+
+        Mono<String> deletedEmployee = employeeService.deleteEmployeeById(employeeId);
+
+        StepVerifier.create(deletedEmployee)
+                .expectNext(responseMessage)
+                .verifyComplete();
+    }
+
+
```

Comment:
Something's not right here. Whitespace and position of `}` make this look like we've got extra space and have lost the indentation.